ql/src/java/org/apache/hadoop/hive/ql/ddl/table/lock/show/ShowDbLocksAnalyzer.java

```diff
@@ -47,14 +47,16 @@ public void analyzeInternal(ASTNode root) throws SemanticException {
     String dbName = stripQuotes(root.getChild(0).getText());
     boolean isExtended = (root.getChildCount() > 1);
 
-    HiveTxnManager txnManager = null;
+    boolean useNewLocksFormat;
     try {
-      txnManager = TxnManagerFactory.getTxnManagerFactory().getTxnManager(conf);
+      HiveTxnManager txnManager = TxnManagerFactory.getTxnManagerFactory().getTxnManager(conf);
```

Comment:
Do we create txnManager instance here just to get the value of useNewLocksFormat flag?

Reply:
That's correct. Normally, we should (but I am not 100% sure) have a TxnManager at this point so there is no need to create a new one just to obtain the flag. I pushed commit https://github.com/apache/hive/pull/1509/commits/297882ee80d52689a9cc1c68da9f7580918439bb to try this out.

Reply:
TxnManager is initialised before query compilation and has a session scope. Correct way to access it is via SessionState. Looks like ShowDbLocksAnalyzer was always creating new instance of TxnManager.

Reply:
Did you check the last commit? Do you have something else in mind?

Reply:
LGTM